### DISCUSSION TITLE
Federated Query Syntax

### DISCUSSION THREAD

**Discussion by juupje** [juupje/pyMogwai#31] [Ideas]
Example for a SPARQL Federated Query:
https://snapquery.bitplan.com/query/bitplan.com/federated-queries/ATMs%20around%20Munich%20belonging%20to%20the%20Bankcard-Servicenetz%20interbank%20network%20(federated%20query)

The goal is to create three named parameterized queries (NPQs); two of them are SPARQL queries and the third is technology agnostic (Gremlin, Mogwai, SPARQL or SQL). The assumption is that each NPQ produces truly tabular, relation algrebra compatible results.  Meaning that the core functionality is a JOIN operation. The question is whether general subgraph joining is feasible or desirable. For a start, being able to do a subset of relational algebra compatible query JOIN would already be very useful.

**Comment by juupje** [comment]
The SPARQL syntax is not straightforward, as the existence of a operation JOIN is lost to non-experts.

**Comment by WolfgangFahl** [comment]
https://cr.bitplan.com/index.php/Technology_Agnostic_Join
But see also
[CORBA Standard IDL](https://de.wikipedia.org/wiki/Common_Object_Request_Broker_Architecture)

```
3.13 Operation Declaration
Operation declarations in OMG IDL are similar to C function declarations. The syntax
is:
(87) <op_dcl> ::= [ <op_attribute> ] <op_type_spec>
<identifier> <parameter_dcls>
[ <raises_expr> ] [ <context_expr> ]
(88) <op_attribute> ::= “oneway”
(89) <op_type_spec> ::= <param_type_spec>
| “void
```
